File: Data Structures & Algorithms/meeting-schedule-ii/submission-1.py
"""
Definition of Interval:
class Interval(object):
    def __init__(self, start, end):
        self.start = start
        self.end = end
"""

class Solution:
	def minMeetingRooms(self, intervals: List[Interval]) -> int:
		starts, ends = [], []
		for interval in intervals:
			starts.append(interval.start)
			ends.append(interval.end)
		
		starts.sort()
		ends.sort()

		s, e, count, maxRoom = 0, 0, 0, 0

		while s < len(starts) and e < len(ends):
			if starts[s] < ends[e]:
				count += 1
				maxRoom = max(maxRoom, count)
				s += 1
			else:
				count -= 1
				e += 1
		return maxRoom

File: Data Structures & Algorithms/merge-intervals/submission-1.go
func merge(intervals [][]int) [][]int {
    res := make([][]int, 0, len(intervals))
    sort.Slice(intervals, func(i, j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
    res = append(res, intervals[0])

    for i := 1; i < len(intervals); i++ {
        if res[len(res)-1][1] >= intervals[i][0] {
            res[len(res)-1][1] = max(res[len(res)-1][1], intervals[i][1])
            continue
        }
        res = append(res, intervals[i])
    }
    return res
}
